### ISSUE TITLE
Wrong error: `expected &{unknown}, found Role`

### ISSUE THREAD

**Issue by wangzhen0518** [rust-lang/rust-analyzer#22346]
**rust-analyzer version**: rust-analyzer version: 0.3.2896-standalone (e266f5cab8 2026-05-10)

**rustc version**: rustc 1.95.0 (59807616e 2026-04-14)

**editor or extension**: VSCode

**relevant settings**: Unrelated to this issue.

**repository link (if public, optional)**: None

**code snippet to reproduce**:
```rust
use ambassador::{Delegate, delegatable_trait};

#[delegatable_trait]
trait Run {
    fn roll(&self) {}
}

struct Role;
impl Run for Role {}

#[derive(Delegate)]
#[delegate(Run)]
enum Dango {
    Role(Role),
}
```
rust analyzer gives this error:
```json
[{
	"resource": ...,
	"owner": "rust-analyzer",
	"code": {
		"value": "E0308",
		"target": {
			"$mid": 1,
			"path": "/stable/error_codes/E0308.html",
			"scheme": "https",
			"authority": "doc.rust-lang.org"
		}
	},
	"severity": 8,
	"message": "expected &{unknown}, found Role",
	"source": "rust-analyzer",
	"startLineNumber": 11,
	"startColumn": 3,
	"endLineNumber": 11,
	"endColumn": 19,
	"modelVersionId": 1,
	"origin": "extHost1"
}]
```

But I can run this code with `cargo run` successfully. I used `cargo expand` and got the following code, hope it is helpful.
```rust
use ambassador::{Delegate, delegatable_trait};
trait Run {
    fn roll(&self) {}
}
#[doc(inline)]
///A macro to be used by [`ambassador::Delegate`] to delegate [`Run`]
use _ambassador_impl_Run as ambassador_impl_Run;
#[doc(hidden)]
#[allow(non_snake_case)]
mod ambassador_impl_Run {}
struct Role;
impl Run for Role {}
#[delegate(Run)]
enum Dango {
    Role(Role),
}
#[allow(non_snake_case)]
mod ambassador_module_Run_for_Dango {
    use super::*;
    #[doc(hidden)]
    #[allow(non_camel_case_types)]
    pub trait MatchRun<ambassador_X: Run>: Run {}
    #[allow(non_camel_case_types)]
    impl<ambassador_X: Run, ambassador_Y: Run> MatchRun<ambassador_X>
    for ambassador_Y {}
    impl Run for Dango
    where
        Role: Run,
    {
        #[inline]
        #[allow(unused_braces)]
        fn roll(&self) {
            match self {
                Dango::Role(inner) => return Run::roll(inner),
            }
        }
    }
}
```

**Comment by wangzhen0518** [comment]
Same issue as #21837.
